frontend/src/metabase/admin/databases/components/DatabaseTableEditingSection.tsx

```diff
@@ -0,0 +1,76 @@
+import { useState } from "react";
+import { t } from "ttag";
+
+import Toggle from "metabase/core/components/Toggle";
+import { getResponseErrorMessage } from "metabase/lib/errors";
+import { PLUGIN_DATA_EDITING } from "metabase/plugins";
+import { Box, Flex } from "metabase/ui";
+import type Database from "metabase-lib/v1/metadata/Database";
+import type { DatabaseData, DatabaseId } from "metabase-types/api";
+
+import { DatabaseInfoSection } from "./DatabaseInfoSection";
+import {
+  Description,
+  Error,
+  Label,
+} from "./DatabaseModelFeaturesSection/ModelFeatureToggles.styled";
+
+export function DatabaseTableEditingSection({
+  database,
+  updateDatabase,
+}: {
+  database: Database;
+  updateDatabase: (
+    database: { id: DatabaseId } & Partial<DatabaseData>,
+  ) => Promise<void>;
+}) {
+  const isEditingDatabase = !!database.id;
+  const showTableEditingSection =
+    isEditingDatabase &&
+    database.supportsActions() &&
+    PLUGIN_DATA_EDITING.isEnabled();
+
+  const [error, setError] = useState<string | null>(null);
+
+  const handleToggle = async (enabled: boolean) => {
+    try {
+      setError(null);
+
+      await updateDatabase({
+        id: database.id,
+        settings: { "database-enable-table-editing": enabled },
+      });
+    } catch (err) {
+      setError(getResponseErrorMessage(err) || t`An error occurred`);
+    }
+  };
+
+  if (!showTableEditingSection) {
+    return null;
+  }
+
+  return (
+    <DatabaseInfoSection
+      name={t`Editable tables`}
+      description={t`Allow the data within tables of this database to be edited by Admin users.`}
```

Comment:
!nit the section title matching the label looks a bit weird. Maybe we can have the writing team weigh in?